File: 0053-maximum-subarray/0053-maximum-subarray.py
import sys
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        maxval = -sys.maxsize-1
        summ = 0
        for i in range(len(nums)):
            summ+=nums[i]
            maxval = max(maxval, summ)
            if summ<0:
                summ = 0
        return maxval

File: 0053-maximum-subarray/0053-maximum-subarray.java
// T- O(n)
// S- O(1)
// kadane's algo - maximum subarray sum
class Solution {
    public int maxSubArray(int[] nums) {
        int summ = 0;
        int maxx = Integer.MIN_VALUE;
        for (int i = 0; i < nums.length; i++) {
            int ele = nums[i];
            summ += ele;
            maxx = Math.max(maxx, summ);
            if (summ < 0)
                summ = 0;
        }
        return maxx;
    }
}